mslice/models/workspacemanager/workspace_algorithms.py

```diff
@@ -106,9 +104,12 @@ def _get_property_from_history(name, history):
     return None
 
 
-def get_q_limits(theta, emax, efix):
-    qmin, qmax, qstep = tuple(np.sqrt(E2q * 2 * efix * (1 - np.cos(theta)) * meV2J) / m2A)
-    qmax = np.sqrt(E2q * (2 * efix + emax - 2 * np.sqrt(efix * (efix + emax)) * np.cos(theta[1])) * meV2J) / m2A
+def get_q_limits(theta, en, efix):
+    #calculates the Q(E) line for the given two theta and then finds the min and max values
+    qlines = [np.sqrt(E2q * (2 * efix - en - 2 * np.sqrt(efix * (efix - en)) * np.cos(tth)) * meV2J) / 1e10 for tth in theta[:2]]
+    qmin = np.nanmin(qlines[0])
+    qmax = np.nanmax(qlines[1])
+    qstep = np.sqrt(E2q * 2 * efix * (1 - np.cos(theta[2])) * meV2J) / m2A
```

Comment:
The default q step is too small if you have many detectors. I would choose something like (qmax-qmin)/100, or something independent of detector spacing

Reply:
This was what we had before but it was found to be too coarse, hence we decided to go with the detector spacings, to be more physical. The idea is that the default value is the minimum and if people want to have coarser binning they can change the values themselves.

I would rather not have a step which depends on just a difference between `qmax` and `qmin` exactly because these values don't have anything to do with the detector spacings - e.g. it would give different step sizes for e.g. ARCS/MERLIN vs SEQUOIA/MAPS.

Reply:
@mducle I added an if check for when there are more than 1000 detectors. Does that make sense or should I revert it back to just the spacing?